include/zephyr/input/input.h

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_INPUT_H_
+#define ZEPHYR_INCLUDE_INPUT_H_
+
+/**
+ * @brief Input Interface
+ * @defgroup input_interface Input Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <stdint.h>
+#include <zephyr/device.h>
+#include <zephyr/dt-bindings/input/input-event-codes.h>
+#include <zephyr/kernel.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Input event structure.
+ *
+ * This structure represents a single input event, for example a key or button
+ * press for a single button, or an absolute or relative coordinate for a
+ * single axis.
+ */
+struct input_event {
+	/** Device generating the event or NULL. */
+	const struct device *dev;
+	/** Event type, one of INPUT_EV_*. */
+	uint16_t type;
+	/** Event code, one of INPUT_{KEY,BTN,ABS,REL,...)_*. */
+	uint16_t code;
+	/** Event value. */
+	int32_t value;
+};
+
+/**
+ * @brief Report a new input event.
+ *
+ * This causes all the listeners for the specified device to be triggered,
+ * either synchronously or through the input thread if utilized.
+ *
+ * @param dev Device generating the event or NULL.
+ * @param type Event type, one of INPUT_EV_*.
+ * @param code Event code, one of INPUT_{KEY,BTN,ABS,REL,...)_*.
+ * @param value Event value.
+ * @param sync Set the synchronization bit for the event.
+ * @param timeout Timeout for reporting the event, ignored if
+ *                `CONFIG_INPUT_THREAD=n`.
+ * @return 0 if the message has been processed, a negative error code if
+ *         `CONFIG_INPUT_THREAD=y` and the message failed to be enqueued,
+ *         -EINVAL if type is outside of the allowed range.
```

Comment:
use `@retval` entries here, also use `@kconfig{CONFIG_INPUT_THREAD} is enabled`

Reply:
done